File: hu/codes/java/chapter_stack_and_queue/stack.java
/**
 * File: stack.java
 * Created Time: 2022-11-25
 * Author: krahets (krahets@163.com)
 */

package chapter_stack_and_queue;

import java.util.*;

public class stack {
    public static void main(String[] args) {
        /* Verem tetején lévő elem elérése */
        Stack<Integer> stack = new Stack<>();

        /* Elemek verembe rakása */
        stack.push(1);
        stack.push(3);
        stack.push(2);
        stack.push(5);
        stack.push(4);
        System.out.println("Stack stack = " + stack);

        /* Lista visszaadása nyomtatáshoz */
        int peek = stack.peek();
        System.out.println("Stack top element peek = " + peek);

        /* Elem kivétele a veremből */
        int pop = stack.pop();
        System.out.println("Pop element pop = " + pop + ", after pop, stack = " + stack);

        /* Verem hosszának lekérése */
        int size = stack.size();
        System.out.println("Stack length size = " + size);

        /* Üresség ellenőrzése */
        boolean isEmpty = stack.isEmpty();
        System.out.println("Stack is empty = " + isEmpty);
    }
}

File: hu/codes/python/chapter_stack_and_queue/stack.py
"""
File: stack.py
Created Time: 2022-11-29
Author: Peng Chen (pengchzn@gmail.com)
"""

"""Fő kód"""
if __name__ == "__main__":
    # Verem inicializálása
    # A Pythonban nincs beépített verem osztály, listát használunk veremként
    stack: list[int] = []

    # Elemek betolása a verembe
    stack.append(1)
    stack.append(3)
    stack.append(2)
    stack.append(5)
    stack.append(4)
    print("verem =", stack)

    # A verem tetején lévő elem elérése
    peek: int = stack[-1]
    print("Verem teteje elem peek =", peek)

    # Elem kivétele a veremből
    pop: int = stack.pop()
    print("Kivett elem pop =", pop)
    print("Pop után verem =", stack)

    # A verem hosszának lekérése
    size: int = len(stack)
    print("Verem hossza size =", size)

    # Ellenőrzi, hogy üres-e
    is_empty: bool = len(stack) == 0
    print("A verem üres-e =", is_empty)
